dbt/include/global_project/macros/materializations/view/bigquery_view.sql

```diff
@@ -14,26 +14,20 @@
 
   {{ run_hooks(pre_hooks) }}
 
-  -- drop if exists
-  {%- if old_relation is not none -%}
-    {%- if old_relation.is_table and not flags.FULL_REFRESH -%}
+  -- If there's a table with the same name and we weren't told to full refresh,
+  -- that's an error. If we were told to full refresh, drop it.
+  {%- if old_relation is not none and old_relation.is_table -%}
+    {%- if flags.FULL_REFRESH and not non_destructive_mode -%}
+      {{ adapter.drop_relation(old_relation) }}
+    {%- else -%}
       {{ exceptions.relation_wrong_type(old_relation, 'view') }}
     {%- endif -%}
-
-    {{ adapter.drop_relation(old_relation) }}
   {%- endif -%}
 
   -- build model
-  {% if exists_as_view and non_destructive_mode -%}
-    {% call noop_statement('main', status="PASS", res=None) -%}
-      -- Not running : non-destructive mode
-      {{ sql }}
-    {%- endcall %}
-  {%- else -%}
-    {% call statement('main') -%}
-      {{ create_view_as(target_relation, sql) }}
-    {%- endcall %}
-  {%- endif %}
+  {% call statement('main') -%}
```

Comment:
I think we might want to keep the logic skip building the view in `--non-destructive` mode. Using `--non-destructive` is generally a bad idea on BigQuery, as this flag is specifically intended to avoid issues with `drop..cascade` on Redshift.

I'm very, very open to removing the flag altogether! But I think we should do that in a separate PR. I suppose I am open to implementing the logic show in this diff (ie. ignoring `--non-destructive` on BigQuery) if we also add a big warning message at the start of the run indicating that `--non-destructive` is ignored on BQ